pyeffects/Either.py

```diff
@@ -85,7 +85,16 @@ def __init__(self, value: A) -> None:
 
     def left(self):
         return self.value
-
+    
+    def __eq__(self, __value: object) -> bool:
```

Comment:
why the double underscore?  can we just rename this to `other`.  Also, the type should be `Either[A]` or something like that.

Reply:
i.e.,

`def __eq__(self, other: Either[A]) -> bool:`